add Builtin/yamlgui, with demo

erladev opened the pull request:
Adding the yamlgui system,  for creating LUI objects from an structured yaml file.
Current features:
- most (probably all) LUI Classes supported
- Keyword Arguments
- nested structure, template instancing from python code
- python code inside the yaml file
- event binding
- removing / changing UI using a single function

Commit:
add Builtin/yamlgui, with demo

rdb commented:
A few suggestions...
- Doesn't seem to be Python 3 compatible due to use of various deprecated syntax (such as implicit relative import, "exec in", etc.)
- Inconsistent use of tabs/spaces, I would suggest sticking to 4 spaces for indentation as is the de facto standard.
- Why define both getitem and getattr to the same thing?  It seems redundant.
- Naming convention seems inconsistent, class names should use CamelCase and methods should use snake_case.
- LUIWidgeds should probably be spelled LUIWidgets
- Spacing seems inconsistent, I would suggest one blank line between methods, and never more than two blank lines
- Calling gc.collect() from a destructor seems like an altogether bad idea to me.  Is it necessary to call it at all?

Commit:
Fixed spacing, naming convention, and use of tabs
Removed yaml-python scripting, handlers are implemented as normal methods
Added possibility to remove a instanced branch again, given the same name
and formatting list as in instance_element
